pkg/vulnerabilityreport/builder.go

```diff
@@ -129,6 +129,11 @@ func (s *ScanJobBuilder) Get() (*batchv1.Job, []*corev1.Secret, error) {
 			},
 		},
 	}
+	// secrets will be created with scan jobs in same namespace where scan job will run
+	for i, _ := range secrets {
+		secrets[i].Namespace = s.pluginContext.GetNamespace()
+	}
```

Comment:
Isn't this code duplicated within the updateScanJobForWorkloadNamespace method?

Reply:
This is existing code that was there in [pkg/operator/controller/vulnerabilityreport.go](https://github.com/aquasecurity/starboard/pull/943/files#diff-6e57a58bb4c7202d8e6d177078a99e7d95b3e436dd3f4e0e90eb990a1e2e4987L262) Due to that code update in updateScanJobForWorkloadNamespace method was not reflecting at the time of secret creation.

So this namespace set is required regardless of sameNamespace option, so that secrets are created in starboad-operator ns. We can do this inside plugin, but since builder is setting namespace for job, so setting it here.